core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+                    'received: \'%s\'' % tag)
+
+            if (tag[0] not in string.ascii_lowercase or
+                    tag[-1] not in string.ascii_lowercase):
+                raise utils.ValidationError(
+                    'Tags should not start or end with whitespace, received: '
+                    '\'%s\'' % tag)
+
+            if re.search(r'\s\s+', tag):
+                raise utils.ValidationError(
+                    'Adjacent whitespace in tags should be collapsed, '
+                    'received: \'%s\'' % tag)
+
+        if strict:
+            if len(tags) == 0:
+                raise utils.ValidationError(
+                    'Atleast one tag should be selected')
+
+        if len(set(tags)) != len(tags):
+            raise utils.ValidationError('Some tags duplicate each other')
+
+    def to_dict(self):
+        """Returns a dict representing this blog post summary domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of blog post instance.
+        """
+        return {
+            'id': self.id,
+            'author_name': user_services.get_username(
+                self.author_id),
+            'title': self.title,
+            'summary': self.summary,
+            'thumbnail_filename': self.thumbnail_filename,
+            'tags': self.tags,
+            'url_fragment': self.url_fragment,
+        }
+
+    def serialize(self):
+        """Returns the object serialized as a JSON string.
+
+        Returns:
+            str. JSON-encoded utf-8 string encoding all of the information
+            composing the object.
+        """
+        blog_post_summary_dict = self.to_dict()
+
+        if self.last_updated:
+            blog_post_summary_dict['last_updated'] = (
+                utils.convert_naive_datetime_to_string(self.last_updated))
+
+        if self.published_on:
+            blog_post_summary_dict['published_on'] = (
+                utils.convert_naive_datetime_to_string(self.published_on))
+
+        return json.dumps(blog_post_summary_dict).encode('utf-8')
+
+
+class BlogPostRights(python_utils.OBJECT):
+    """Domain object for Blog Post rights."""
+
+    def __init__(self, blog_post_id, editor_ids, blog_post_is_published=False):
+        """Constructs a BlogPostRights domain object.
+
+        Args:
+            blog_post_id: str. The id of the blog post.
+            editor_ids: list(str). The id of the users who have been assigned
+                as editors for the blog post.
+            blog_post_is_published: bool. Whether the blog is published or not.
+        """
+        self.id = blog_post_id
+        self.editor_ids = editor_ids
+        self.blog_post_is_published = blog_post_is_published
+
+    def to_dict(self):
+        """Returns a dict suitable for use by the frontend.
+
+        Returns:
+            dict. A dict version of BlogPostRights suitable for use by the
+            frontend.
+        """
+        return {
+            'blog_post_id': self.id,
```

Comment:
This should match the field name, right?

Reply:
To remove confusion in front-end as the ID of the BlogPostRights is also the same as the ID of all the models of Blog Post, and will be used to fetch the complete blog post, we named it 'blog_post_id' instead of 'id'.